src/ServiceManager.php

```diff
@@ -239,27 +229,24 @@ public function get($name)
 
         // At this point, we have to create the object.
         // We use the resolved name for that.
-        $object = $this->doCreate($resolvedName);
+        /** @psalm-suppress MixedAssignment Yes indeed, service managers can return mixed. */
+        $service = $this->doCreate($resolvedName);
 
         // Cache the object for later, if it is supposed to be shared.
         if ($sharedService) {
-            $this->services[$resolvedName] = $object;
+            $this->services[$resolvedName] = $service;
+            $this->services[$name]         = $service;
         }
 
-        // Also cache under the alias name; this allows sharing based on the
-        // service name used.
-        if ($sharedAlias) {
-            $this->services[$name] = $object;
-        }
```

Comment:
Removed on purpose? :thinking: 

Alias caching was kinda relevant for performance

Reply:
There is a check if it is a shared alias a few lines above and thus the if was redundant. Just moved the caching to another if which should suffice.

Reply:
Ah, I see that indeed you added some code around `if ($sharedService) {` :+1: